### PR TITLE
Split MakeCSR for new signature.

### PR DESCRIPTION
**Pull request by jcbsmpsn**

### CODE REVIEW

**Comment by jcbsmpsn** [comment]
cc @deads2k

**Comment by deads2k** [comment]
@liggitt you're closer to this spot than I am.

File: staging/src/k8s.io/client-go/util/cert/csr.go [hunk 1]
```diff
@@ -28,36 +28,48 @@ import (
...
+		switch {
+		case keySize >= 4096:
+			return x509.SHA512WithRSA
+		case keySize >= 3072:
+			return x509.SHA384WithRSA
+		default:
+			return x509.SHA256WithRSA
+		}
+	}
+	return x509.UnknownSignatureAlgorithm
```
Comment (deads2k) [inline staging/src/k8s.io/client-go/util/cert/csr.go]: @liggitt this bit in particular looks like the functional change.  Used to be the zero value, now its explicitly unknown, which I *think* is the same thing.

Comment (mikedanese) [inline staging/src/k8s.io/client-go/util/cert/csr.go]: @deads2k x509.UnknownSignatureAlgorithm is the zero value.

**Pull request merged**